src/test/java/org/wso2/carbon/apimgt/securityenforcer/publisher/sync/SyncPublisherThreadPoolTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.apimgt.securityenforcer.publisher.sync;
+
+import org.junit.Test;
+import org.wso2.carbon.apimgt.securityenforcer.dto.AISecurityHandlerConfig;
+import org.wso2.carbon.apimgt.securityenforcer.internal.ServiceReferenceHolder;
+
+public class SyncPublisherThreadPoolTest {
+
+    @Test
+    public void getSyncPublisherThreadPoolInstance() throws Exception {
+        AISecurityHandlerConfig aiSecurityHandlerConfig = new AISecurityHandlerConfig();
+        AISecurityHandlerConfig.StackObjectPoolConfig stackObjectPoolConfig = new AISecurityHandlerConfig.StackObjectPoolConfig();
+        stackObjectPoolConfig.setMaxIdle(10);
+        stackObjectPoolConfig.setInitIdleCapacity(10);
+        aiSecurityHandlerConfig.setStackObjectPoolConfig(stackObjectPoolConfig);
+        ServiceReferenceHolder.getInstance().setSecurityHandlerConfig(aiSecurityHandlerConfig);
+        SyncPublisherThreadPool dataPublisherPoolInstance = SyncPublisherThreadPool.getInstance();
+        SyncPublishingAgent callablePublishingAgent = dataPublisherPoolInstance.get();
+        dataPublisherPoolInstance.release(callablePublishingAgent);
+        dataPublisherPoolInstance.cleanup();
```

Comment:
Needs to have some kind of verification logic to consider this as a test. Currently seems to be only executing code.